lib/spack/spack/package_base.py

```diff
@@ -2924,14 +2924,21 @@ class BundlePackage(PackageBase):
 
     #: There are no phases by default but the property is required to support
     #: post-install hooks (e.g., for module generation).
-    phases = []  # type: List[str]
+    phases = ["promote"]  # type: List[str]
     #: This attribute is used in UI queries that require to know which
     #: build-system class we are using
     build_system_class = "BundlePackage"
 
     #: Bundle packages do not have associated source or binary code.
     has_code = False
 
+    def promote(self, spec, prefix):
+        # Update the explicit dependencies
+        for _, cond_dict in self.dependencies.items():
+            for cond, dep in cond_dict.items():
+                if dep.explicit and cond in spec:
+                    spack.store.db.update_explicit(spec[dep.spec.name], True)
+
```

Comment:
How about:

```suggestion
    # list of direct dependency names that should be marked explicit after install
    explicit = []
    
    @run_after("install")
    def mark_explicit(self):
        for dep in self.spec.dependencies():
            if dep.name in BundlePackage.explicit:
                spack.store.db.update_explicit(dep, True)
```

Reply:
Is there a way to get_explicit from the database of a package in `run_after("install")` I don't think this is updated until after the install step of the pacakge is complete but I couldn't quite understand the order.

We don't want to promote the bundle deps to a higher level than their parent bundle package.